File: docs/source/_src_snippets/simulation/environment_setup/environment_models/custom_constant_temperature_atmosphere.py
density_function = ...
constant_temperature = 250.0
specific_gas_constant = 300.0
ratio_of_specific_heats = 1.4

body_settings.get( "Earth" ).atmosphere_settings = environment_setup.atmosphere.custom_constant_temperature( 
    density_function, constant_temperature, specific_gas_constant, ratio_of_specific_heats)

File: docs/source/_snippets/simulation/environment_setup/environment_models/custom_constant_temperature_atmosphere.cpp
// Outside main
double customDensityFunction( const double altitude, const double longitude, const double latitude, const double time )
{
   // Return a linear combination of the input values
   return 0.5 * altitude + 0.25 * longitude + 0.15 * latitude + 0.1 * time;
}

int main( )
{
  // ...

  // Define atmosphere settings
  double constantTemperature = 250.0;
  double specificGasConstant = 300.0;
  double ratioOfSpecificHeats = 1.4;
  bodySettings[ "Earth" ]->atmosphereSettings = std::make_shared< CustomConstantTemperatureAtmosphereSettings >( 
                                      &customDensityFunction, constantTemperature, specificGasConstant, ratioOfSpecificHeats );

  // ...
}